vmm/src/device_manager.rs

```diff
@@ -895,6 +984,40 @@ impl DeviceManager {
         Ok(device_manager)
     }
 
+    pub fn serial_pty(&self) -> Option<File> {
+        if self.serial_pty.is_some() {
```

Comment:
You *could* write this as `self.serial_pty.as_ref().map(|pty| pty.unwrap().lock().unwrap().0.try_clone().unwrap())`

We normally just simply panic on unlock() or try_clone() errors. There is nothing the user can really do about it at runtime.

Reply:
Ah okay, I thought I had copied the `lock().expect()` from another usage in the code. Noted for `try_clone()` checking as well, thanks!

Reply:
Fixed.